code/src/components/app-dropdown/AppDropdown.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useRef, MouseEvent, FocusEvent, ReactNode } from "react";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+import AppDropdownOptions from "@/components/app-dropdown/components/AppDropdownOptions";
+import { AppDropdownProps } from "@/components/app-dropdown/AppDropdown.types";
+
+import "@/components/app-dropdown/AppDropdown.scss";
+
+const AppDropdown = ({
+  options,
+  onSelect,
+  defaultLabel,
+  className
+}: AppDropdownProps) => {
+  const selectedLabelValue = defaultLabel || options[0].label;
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedLabel, setSelectedLabel] =
+    useState<ReactNode>(selectedLabelValue);
+  const dropdownRef = useRef<HTMLElement>(null);
+
+  const iconConditional = isOpen ? (
+    <KeyboardArrowUpIcon fontSize="medium" />
+  ) : (
+    <KeyboardArrowDownIcon fontSize="medium" />
+  );
+
+  const handleSelect = (value: string, label: ReactNode) => {
+    setSelectedLabel(label);
+    setIsOpen(false);
+    onSelect(value);
+  };
+
+  const handleToggleDropdownClick = () => {
+    setIsOpen((prevOpenState) => !prevOpenState);
+  };
+
+  const handleOptionClick = (
+    event: MouseEvent,
+    value: string,
+    label: ReactNode
+  ) => {
+    event.stopPropagation();
+    handleSelect(value, label);
+  };
+
+  const handleBlur = (event: FocusEvent<HTMLElement>) => {
+    if (!dropdownRef.current?.contains(event.relatedTarget as Node)) {
```

Comment:
Can you please explain why do we use `relatedTarget` here?

Reply:
> Can you please explain why do we use `relatedTarget` here?

It's used to determine if focus has moved outside the dropdown component
Here is the article can read about it:
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/relatedTarget

Reply:
Interesting solution. Great work)